internal/resolvers/youtube/helpers.go

```diff
@@ -3,9 +3,27 @@ package youtube
 import (
 	"net/url"
 	"path"
+	"regexp"
 	"strings"
 )
 
+type ChannelType string
+const (
+	// InvalidChannel channel isn't of a known type or doesn't exist
+	InvalidChannel ChannelType = ""
+	// UserChannel channel ID is a username
+	UserChannel = "user"
+	// IdentifierChannel channel uses the YouTube channel ID format (UC*)
+	IdentifierChannel = "channel"
+	// CustomChannel channel uses a custom URL and requires a Search call for the ID
+	CustomChannel = "c"
+)
+
+type ChannelId struct {
+	id string
+	channelType ChannelType
+}
```

Comment:
Shouldn't this be moved to `model.go`?

Reply:
I sorta see `helpers.go` as the 'owner' of that type but happy to move it into model to align it.